diff --git a/conf/afm-user-daemon.service b/conf/afm-user-daemon.service
index 6b5c1d8..e0f6799 100644
--- a/conf/afm-user-daemon.service
+++ b/conf/afm-user-daemon.service
@@ -7,6 +7,7 @@ BusName=org.AGL.afm.user
 ExecStart=/usr/bin/afm-user-daemon --user-dbus=unix:path=%t/bus
 Environment=AFM_APP_INSTALL_DIR=%%r
 Environment=LD_PRELOAD=/usr/lib/libEGL.so
+Environment=QT_WAYLAND_SHELL_INTEGRATION=ivi-shell
 Restart=on-failure
 RestartSec=5
