diff --git a/conf/afm-unit.conf b/conf/afm-unit.conf
index 82113ef..2fbc9e2 100644
--- a/conf/afm-unit.conf
+++ b/conf/afm-unit.conf
@@ -127,6 +127,7 @@ SuccessExitStatus=0 SIGKILL
 WorkingDirectory=-{{&#metadata.app-data-dir}}/{{id}}
 ExecStartPre=/bin/mkdir -p {{&#metadata.app-data-dir}}/{{id}}
 Environment=AFM_APP_INSTALL_DIR={{:#metadata.install-dir}}
+Environment=QT_WAYLAND_SHELL_INTEGRATION=ivi-shell
 
 %systemd-unit user
 {{#required-permission.urn:AGL:permission::public:hidden}}\
