From 8f761e02172544212915c82b7e8dd8d4dd1281a6 Mon Sep 17 00:00:00 2001
From: Petr Nechaev <petr.nechaev@cogentembedded.com>
Date: Mon, 3 Aug 2015 14:18:34 +0300
Subject: [PATCH 9/9] Add 'Pressure' property to 'BrakeOperation'

---
 plugins/dbus/runningstatus.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/dbus/runningstatus.h b/plugins/dbus/runningstatus.h
index 16321cb..a706528 100644
--- a/plugins/dbus/runningstatus.h
+++ b/plugins/dbus/runningstatus.h
@@ -115,6 +115,7 @@ public:
 		:DBusSink("BrakeOperation", re, connection, map<string, string>())
 	{
 		wantPropertyVariant(VehicleProperty::WheelBrake, "BrakePedalDepressed", VariantType::Read);
+		wantPropertyVariant(VehicleProperty::WheelBrakePressure, "BrakePressure", VariantType::Read);
 	}
 };
 
-- 
1.9.1

